fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/NamespaceTools.java

```diff
@@ -90,6 +94,12 @@ public static void validatePath(final Session session, final String path) {
         final String relPath = path.replaceAll("^/+", "").replaceAll("/+$", "");
         final String[] pathSegments = relPath.split("/");
         for (final String segment : pathSegments) {
+            // Empty path elements are not valid!
```

Comment:
No, those locations are correct usage of this method, and the method is fine. You are implementing the new test in the wrong place. It should not be in this method. This method processes JCR paths, but you have been asked to process an HTTP URI. The new test should be in `HttpResourceConverter` and nowhere else, and it should not be used anywhere else. You should also refactor `FedoraLdp` and `FedoraNodes` to avoid using this method, and instead to use a validation strategy based on the use of an `IdentifierConverter`.

Consistency in error is no virtue.